bindings/rust/build.rs

```diff
@@ -30,38 +21,85 @@ fn main() {
 
     eprintln!("Using FIELD_ELEMENTS_PER_BLOB={}", field_elements_per_blob);
 
-    // Deleting any existing assembly and object files to ensure that compiling with a different
-    // feature flag changes the final linked library file.
-    let obj_file = root_dir.join("src").join("c_kzg_4844.o");
-    if obj_file.exists() {
-        std::fs::remove_file(obj_file).unwrap();
+    let c_src_dir = root_dir.join("src");
+
+    let mut cc = cc::Build::new();
+
+    let file_vec = vec![c_src_dir.join("c_kzg_4844.c")];
+
+    /*
+     * BLST env
+     */
+
+    let target_arch = env::var("CARGO_CFG_TARGET_ARCH").unwrap();
+
+    if target_arch.eq("x86_64") || target_arch.eq("aarch64") {
+    } else {
+        cc.define("__BLST_NO_ASM__", None);
     }
+    match (cfg!(feature = "portable"), cfg!(feature = "force-adx")) {
+        (true, false) => {
+            println!("Compiling in portable mode without ISA extensions");
+            cc.define("__BLST_PORTABLE__", None);
+        }
+        (false, true) => {
+            if target_arch.eq("x86_64") {
+                println!("Enabling ADX support via `force-adx` feature");
+                cc.define("__ADX__", None);
+            } else {
+                println!("`force-adx` is ignored for non-x86_64 targets");
+            }
+        }
+        (false, false) =>
+        {
+            #[cfg(target_arch = "x86_64")]
+            if target_arch.eq("x86_64") && std::is_x86_feature_detected!("adx") {
+                println!("Enabling ADX because it was detected on the host");
+                cc.define("__ADX__", None);
+            }
+        }
+        (true, true) => panic!("Cannot compile with both `portable` and `force-adx` features"),
+    }
+    /*
+     * END OF BLST env
+     */
 
-    // Ensure libckzg exists in `OUT_DIR`
-    Command::new("make")
-        .current_dir(root_dir.join("src"))
-        .arg("c_kzg_4844.o")
-        .arg(format!(
-            "FIELD_ELEMENTS_PER_BLOB={}",
-            field_elements_per_blob
-        ))
+    // Obtain the header files exposed by blst-bindings' crate.
+    let blst_headers_dir =
+        std::env::var_os("DEP_BLST_BINDINGS").expect("BLST exposes header files for bindings");
+    /*
+     * Hack ahead
+     */
+    const AWK_COMMAND: &str =
+        "{gsub(/typedef struct \\{\\} blst_uniq;/, \"typedef struct { int dummy; } blst_uniq;\")}1";
+    let awk_succeeds = Command::new("awk")
```

Comment:
We should look into ways of removing this hack. On Windows, if you compile with GCC, I don't think this hack will be necessary. I believe this is only a problem with MSVC. But if we want to support both MSVC & GCC, then yeah I think this is necessary.

Reply:
since blst is compiled by the library, the safer option is using the same compiler (cl) instead of gcc

Reply:
Do you think we could submit this change to the blst crate itself instead?

Reply:
I believe that would be the ideal solution. Let's see if we can make that happen.

Reply:
Made a PR here. Not sure if they'll like it or how long it could take.

* https://github.com/supranational/blst/pull/157

Reply:
I suggest we proceed without that blst PR. If it get's merged in the future, we'll remove whatever hack we do here.